modules/workflow-service-impl/src/main/java/org/opencastproject/workflow/impl/WorkflowServiceImpl.java

```diff
@@ -949,11 +980,34 @@ else if (!FAILED.equals(workflow.getState())) {
     return processingOperation;
   }
 
+  private boolean userCanAccessWorkflow(WorkflowDefinition wd) {
+    if (wd.getRoles().isEmpty() || securityService.getUser().getRoles().stream()
+            .anyMatch(role -> role.getName().equals(GLOBAL_ADMIN_ROLE))) {
+      return true;
+    }
```

Comment:
Might be a matter of taste, but I would prefer...

` } else {`

... over ...

```
  return true;
}
... the else case
```